app/models/user.rb

```diff
@@ -20,4 +20,14 @@ def send_registered_email
     RegistrationMailer.registration_email(self).deliver_now
   end
 
+
+  validate :validate_donor_fields
+
+  def validate_donor_fields
+    if role_type == 'Donor'
```

Comment:
Looked at the method definition, it looks like the issue is that it looks for the role to actually be defined (rather than just specifying a type), which it won't be during validations on a new user.

This should be fine to leave as-is, then. :sheep: it